### DISCUSSION TITLE
How to create a command like `:sync_sub` in lf?

### DISCUSSION THREAD

**Discussion by LinArcX** [gokcehan/lf#922] [Ideas]
Hi.

Imaine i have a video called: 
`House.of.the.Dragon.S01E01.WEB.1080p.x265.HEVC.mkv`

And also I have a subtitle:
`House.of.the.Dragon.S01E01.WEB.1080p.x265.HEVC.srt`

The problem is that, subtitle isn't synced with video, so I'm using [alass](https://github.com/kaegi/alass) to make it sync like this:

```
alass House.of.the.Dragon.S01E01.WEB.1080p.x265.HEVC.mkv House.of.the.Dragon.S01E01.WEB.1080p.x265.HEVC.srt output.srt
```

I want to automate this process using **lf**. for example, I want select both video and _unsynced_ subtitle, run a command like `:sync_sub` in lf and the command that you saw earlier should be run in the background.

Also at the end, I want to:
1. rename: `House.of.the.Dragon.S01E01.WEB.1080p.x265.HEVC.srt` to `House.of.the.Dragon.S01E01.WEB.1080p.x265.HEVC.srt_`

2. rename: `output.srt` to `House.of.the.Dragon.S01E01.WEB.1080p.x265.HEVC.srt`

Any idea?

**Comment by DusanLesan** [comment]
Maybe this:
```
cmd sync_sub ${{
	srt="$(echo "$fx" | grep '.srt')"
	mov="$(echo "$fx" | grep -v '.srt')"
	alass "$mov" "$srt" "output.srt"
	mv "$srt" "$srt.bak"
	mv "output.srt" "${mov%.*}.srt"
}}
```
You would need to have movie and srt file selected before giving command

**Reply by LinArcX** [reply]
Thank you so much. It works like a charm :)

**Reply by ilyagr** [reply]
@LinArcX You should officially mark this as an answer so that people know you've been helped.

**Reply by LinArcX** [reply]
I can't. Because this is IDEA not Q/A.
